### ISSUE TITLE
SliceViewer data shifts when thickness is adjusted in non-ortho projection

### ISSUE THREAD

**Issue by DanielMurphy22** [mantidproject/mantid#32317]
Originally reported in Manual testing : https://github.com/mantidproject/mantid/issues/32281#issuecomment-905462572

**Describe the bug**
On Sliceviewer for Nonorthogonal data, I can get data to display in the wrong place (it moves but the axes values do not).

**To Reproduce**
- Run this script:
```
from mantid.simpleapi import *
# SXD23767.raw is available in the TrainingCourseData from the downloads page
SXD23767 = Load(Filename='SXD23767.raw', LoadMonitors='Exclude')
# Set some UB with angles we can play with
SetUB(SXD23767, 1,1,2,90,90,120)
SXD_MD_nonortho = ConvertToDiffractionMDWorkspace(InputWorkspace='SXD23767', OutputDimensions='HKL')
```

- Open sliceviewer on the workspace `SXD_MD_nonortho`. 
- Put K on the X axis and H on the Y, and also press the home toolbar button to reset the axes. 
- Click the trifold Nonortho projection toolbar button. 
<img width="841" alt="Screenshot 2021-08-25 at 13 32 36" src="https://user-images.githubusercontent.com/55980573/130791394-bf763b52-7e30-458f-8bfb-4c68499220da.png">

- Finally either adjust the L axis slider or it's thickness and notice the data has shifted to the left and zoomed a little (but the axis values remain the same). To play around with this alternate clicking the home button and adjusting the thickness.
<img width="845" alt="Screenshot 2021-08-25 at 13 33 02" src="https://user-images.githubusercontent.com/55980573/130791405-1cfae6fd-bc7c-484b-88aa-c223ff5cbeaa.png">

**Expected behavior**
The data does not shift when the L thickness or L slider is adjusted

**Platform/Version (please complete the following information):**
 - OS: IDAaaS (RHEL7) and Windows 10
 - Mantid Version pre-beta release of 6.2, also exists in 6.1 and 5.1.1

**Additional context**
In the pre-beta release of 6.2, this data moving problem exists only in the non-orthogonal view, but in releases 6.1 and 5.1.1, the problem existed in both non-ortho and normal view, so in the last dev cycle we’ve fixed this for normal view somehow.

**Comment by RichardWaiteSTFC** [comment]
Interestingly I can reproduce this with SXD data but not with fake data (Event or histo) made a so
```
ws = CreateMDWorkspace(Dimensions='3', Extents='-6,6,-4,4,-0.5,0.5',
                       Names='H,K,L', Units='r.l.u.,r.l.u.,r.l.u.',
                       Frames='HKL,HKL,HKL',
                       SplitInto='2', SplitThreshold='50')
print(ws.getSpecialCoordinateSystem())
expt_info = CreateSampleWorkspace()
ws.addExperimentInfo(expt_info)
SetUB(ws, 1,1,2,90,90,120)
FakeMDEventData(ws, PeakParams='1e+06,0,0,0,1', RandomSeed='3873875')
 
# try with MDHistoToWorkspace
BinMD(InputWorkspace='ws', AxisAligned=False, 
    BasisVector0='H,r.l.u.,1.0,0.0,0.0', 
    BasisVector1='K,r.l.u.,0.0,1.0,0.0', 
    BasisVector2='L,r.l.u.,0.0,0.0,1.0',
    OutputExtents='-6,6,-4,4,-0.5,0.5', OutputBins='100,100,100', 
    NormalizeBasisVectors=False, OutputWorkspace='ws_histo')
```

**Comment by martyngigg** [comment]
This is important to fix so we're including it in 6.2.

**Comment by RichardWaiteSTFC** [comment]
> This is important to fix so we're including it in 6.2.

@martyngigg can we also move #32314 to 6.2, it can cause an exception and I've already found a fix

**Comment by martyngigg** [comment]
> @martyngigg can we also move #32314 to 6.2, it can cause an exception and I've already found a fix

That's fine. Please create the pull request against `release-next` and then we can assess to risk and move if we feel uncomfortable.

**Comment by martyngigg** [comment]
Fixed by #32375
